### DISCUSSION TITLE
MQTT plugin

### DISCUSSION THREAD

**Discussion by sonicaghi** [Kong/kong#7393] [Ideas and feature requests]
https://eclipse.org/community/eclipse_newsletter/2014/february/article2.php

**Comment by Tieske** [comment]
What would you expect this plugin to do?

**Reply by yaobo-lab** [reply]
transformations. since kong doesnt support this protocol natively.

**Comment by sonicaghi** [comment]
transformations. since kong doesnt support this protocol natively.

**Comment by t1tcrucible** [comment]
+1 (if possible)
We were looking aswell in a solution for this, with an additional service complementary, using introspection for authentication with Kong. But seems like a hard nut to crack.

**Comment by Tieske** [comment]
@t1tcrucible can you explain in more detail what your usecase is? Kong runs http not mqtt. So you envision some protocol conversion?  http <-> mqtt ?

**Comment by sundarseswaran** [comment]
I'm just throwing this out here, see below link, is it possible to implement something similar to this for MQTT protocol right in Kong. 

https://github.com/wolfeidau/rifter
